lib/Vanguard.cpp

```diff
@@ -233,7 +233,7 @@ int main(int argc, char **argv) {
         auto &factory = vanguard::BlockchainDomain::Factory::instance();
         std::vector<vanguard::BlockchainDomain::CompilationUnit *> units;
         units.reserve(modules.size());
-        for(auto &mod : modules) {
+        for(auto mod : modules) {
```

Comment:
nit: this is slightly more readable
```suggestion
        for(auto *mod : modules) {
```